macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,243 @@
+          from anomaly_scores
+          where anomaly_score is not null
+      )
+      select metric_value as value,
+             training_avg as average,
+             case when is_anomalous = TRUE then
+              lag(min_metric_value) over (partition by full_table_name, column_name, metric_name, dimension, dimension_value order by bucket_end)
+             else min_metric_value end as min_value,
+             case when is_anomalous = TRUE then
+              lag(max_metric_value) over (partition by full_table_name, column_name, metric_name, dimension, dimension_value order by bucket_end)
+              else max_metric_value end as max_value,
+             bucket_start as start_time,
+             bucket_end as end_time,
+             is_anomalous,
+             dimension,
+             dimension_value,
+             metric_id
+      from anomaly_scores_with_is_anomalous
+      where upper(full_table_name) = upper({{ elementary.const_as_string(full_table_name) }})
+            and metric_name = {{ elementary.const_as_string(metric_name) }}
+           {%- if column_name %}
+              and upper(column_name) = upper({{ elementary.const_as_string(column_name) }})
+           {%- endif %}
+      order by bucket_end, dimension_value
+  {%- endset -%}
+  {% set test_results_description %}
+      {% if has_anomaly_score %}
+          {{ elementary.insensitive_get_dict_value(elementary_test_row, 'anomaly_description') }}
+      {% else %}
+          Not enough data to calculate anomaly score.
+      {% endif %}
+  {% endset %}
+  {% set test_result_dict = {
+      'id': elementary.insensitive_get_dict_value(elementary_test_row, 'id'),
+      'data_issue_id': elementary.insensitive_get_dict_value(elementary_test_row, 'metric_id'),
+      'test_execution_id': elementary.insensitive_get_dict_value(elementary_test_row, 'test_execution_id'),
+      'test_unique_id': test_unique_id,
+      'model_unique_id': parent_model_unique_id,
+      'detected_at': elementary.insensitive_get_dict_value(elementary_test_row, 'detected_at'),
+      'database_name': database_name,
+      'schema_name': schema_name,
+      'table_name': table_name,
+      'column_name': column_name,
+      'test_type': 'anomaly_detection',
+      'test_sub_type': metric_name,
+      'test_results_description': test_results_description,
+      'other': elementary.insensitive_get_dict_value(elementary_test_row, 'anomalous_value'),
+      'owners': elementary.insensitive_get_dict_value(flattened_test, 'model_owners'),
+      'tags': elementary.insensitive_get_dict_value(flattened_test, 'model_tags'),
+      'test_results_query': test_results_query,
+      'test_name': elementary.insensitive_get_dict_value(flattened_test, 'short_name'),
+      'test_params': elementary.insensitive_get_dict_value(flattened_test, 'test_params'),
+      'severity': elementary.insensitive_get_dict_value(flattened_test, 'severity'),
+      'result_rows': elementary.render_result_rows(anomaly_score_rows)
```

Comment:
Shouldn't we limit `anomaly_score_rows` here (or in `render_result_rows`)? because if I understand correctly in `handle_anomaly_test` we query the rows without limit (which makes sense).

Reply:
The query itself is limited to 5 rows on a dbt test,
for an Elementary test, I'd rather either include all the results (only up to `days_back`),
or just have `null` so that we query with `test_results_query`.